FetchMigration/python/orchestrator.py

```diff
@@ -0,0 +1,60 @@
+import argparse
+import logging
+import os
+import subprocess
+
+import migration_monitor
+import pre_migration
+from migration_monitor_params import MigrationMonitorParams
+from pre_migration_params import PreMigrationParams
+
+
+__DP_EXECUTABLE_SUFFIX = "/bin/data-prepper"
+__PIPELINE_OUTPUT_FILE_SUFFIX = "/pipelines/pipeline.yaml"
+
+
+def run(dp_base_path: str, dp_config_file: str, dp_endpoint: str):
+    dp_exec_path = dp_base_path + __DP_EXECUTABLE_SUFFIX
+    output_file = dp_base_path + __PIPELINE_OUTPUT_FILE_SUFFIX
+    pre_migration_params = PreMigrationParams(dp_config_file, output_file, report=True)
+    logging.info("Running pre-migration steps...\n")
+    pre_migration_result = pre_migration.run(pre_migration_params)
+    if len(pre_migration_result.created_indices) > 0:
+        # Kick off a subprocess for Data Prepper
+        logging.info("Running Data Prepper...\n")
+        proc = subprocess.Popen(dp_exec_path)
+        # Data Prepper started successfully, run the migration monitor
```

Comment:
how do you know that it was successful?  What happens when the code isn't successful?  I'm presuming that it throws, but I don't see a catch block that would explain to the user what has happened or what course of action to take next.

Reply:
Added an item to the backlog to improve this - https://opensearch.atlassian.net/browse/MIGRATIONS-1331